lib/jekyll/filters.rb

```diff
@@ -260,6 +261,23 @@ def where_exp(input, variable, expression)
       end
     end
 
+    # Convert the input into integer
+    #
+    # input - the object string
+    #
+    # Returns the integer value
+    def to_integer(input)
+      return input if input.is_a?(Integer)
+      return 1 if input == true
+      return 0 if input == false
+      begin
+        input.to_i
+      rescue
+        raise ArgumentError,
```

Comment:
It's okay but if this was removed, users would get a ```undefined method `to_i'``` instead.
Will this error message mislead users?